cg/services/sample_service/sample_service.py

```diff
@@ -0,0 +1,39 @@
+from cg.services.sample_service.utils import (
+    get_cancel_comment,
+    get_confirmation_message,
+)
+from cg.store.models import User
+from cg.store.store import Store
+
+
+class SampleService:
+
+    def __init__(self, store: Store):
+        self.store = store
+
+    def cancel_sample(self, sample_id: int, user_email: str) -> None:
+        case_ids: list[str] = self.store.get_case_ids_with_sample(sample_id)
+        self.store.mark_sample_as_cancelled(sample_id)
+        self.store.decouple_sample_from_cases(sample_id)
+        self.store.delete_cases_without_samples(case_ids)
+        self._add_cancel_comment(sample_id=sample_id, user_email=user_email)
+
+    def _add_cancel_comment(self, sample_id: int, user_email: str) -> None:
+        if user := self.store.get_user_by_email(user_email):
+            comment = get_cancel_comment(user.name)
+            self.store.update_sample_comment(sample_id=sample_id, comment=comment)
```

Comment:
Would we want to validate that the user exists before acting on anything at all? Or that is done by the endpoint already right?

Reply:
Yeah, it is!